src/gt4py/next/program_processors/runners/dace_fieldview/gtir_to_tasklet.py

```diff
@@ -167,16 +184,290 @@ def _visit_deref(self, node: gtir.FunCall) -> MemletExpr | ValueExpr:
+                    {new_index_connector},
+                    f"{new_index_connector} = index + {offset_expr}",
+                )
+            else:
+                dynamic_offset_tasklet = self._add_tasklet(
+                    "dynamic_offset",
+                    {"index", "offset"},
+                    {new_index_connector},
+                    f"{new_index_connector} = index + offset",
+                )
+            for input_expr, input_connector in [(index_expr, "index"), (offset_expr, "offset")]:
+                if isinstance(input_expr, MemletExpr):
+                    self._add_entry_memlet_path(
+                        input_expr.node,
+                        input_expr.subset,
+                        dynamic_offset_tasklet,
+                        input_connector,
+                    )
+                elif isinstance(input_expr, ValueExpr):
+                    self.state.add_edge(
+                        input_expr.node,
+                        None,
+                        dynamic_offset_tasklet,
+                        input_connector,
+                        dace.Memlet(data=input_expr.node.data, subset="0"),
+                    )
+
+            if isinstance(index_expr, SymbolExpr):
+                dtype = index_expr.dtype
+            else:
+                dtype = index_expr.node.desc(self.sdfg).dtype
+
+            new_index = self._get_tasklet_result(dtype, dynamic_offset_tasklet, new_index_connector)
+
+        # a new iterator with a shifted index along one dimension
+        return IteratorExpr(
+            it.field,
+            it.dimensions,
+            {dim: (new_index if dim == offset_dim else index) for dim, index in it.indices.items()},
+        )
+
+    def _make_dynamic_neighbor_offset(
+        self,
+        offset_expr: MemletExpr | ValueExpr,
+        offset_table_node: dace.nodes.AccessNode,
+        origin_index: SymbolExpr,
+    ) -> ValueExpr:
+        """
+        Implements access to neighbor connectivity table by means of a tasklet node.
+
+        It requires a dynamic offset value, either obtained from a field/scalar argument (`MemletExpr`)
+        or computed by another tasklet (`ValueExpr`).
+        """
+        new_index_connector = "neighbor_index"
+        tasklet_node = self._add_tasklet(
+            "dynamic_neighbor_offset",
+            {"table", "offset"},
+            {new_index_connector},
+            f"{new_index_connector} = table[{origin_index.value}, offset]",
+        )
+        self._add_entry_memlet_path(
+            offset_table_node,
+            sbs.Range.from_array(offset_table_node.desc(self.sdfg)),
+            tasklet_node,
+            "table",
+        )
+        if isinstance(offset_expr, MemletExpr):
+            self._add_entry_memlet_path(
+                offset_expr.node,
+                offset_expr.subset,
+                tasklet_node,
+                "offset",
+            )
+        else:
+            self.state.add_edge(
```

Comment:
I am a bit curious, you have a `_add_tasklet()` function but why do you not have an `add_edge()` function.
As far as I can tell, they are just wrapper around DaCe functions.

Reply:
The `_add_tasklet()` function uses `self.state`. You are right, it is good to do the same for `add_edge() `

Reply:
I forgot to mention the main reason for `_add_map()` and `_add_tasklet()`. They do not call the dace method directly, but another wrapper that assigns a unique name. Edges do not require a name. Anyway, I will add it for consistency.